### DISCUSSION TITLE
how i can show the Term Display name inside the PnP search result

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#2717] [Q&A]
I am using the latest version of PnP-modern-search. We have a SharePoint column of type managed metadata named "MainTypeTest". now i am showing it inside the PnP Search Result web part as follow using the managed property named "owstaxIdMainTypeTermTest":-

![image](https://user-images.githubusercontent.com/79726511/214741725-2a5e6014-c46d-44bd-95e6-0a39e99de184.png)

![image](https://user-images.githubusercontent.com/79726511/214741764-b5bc014d-c68b-4814-95f2-fa6e5b423f42.png)

but we are getting the technical name of the term and not the user firendly name which is "Process" in our case.
any advice ?

![image](https://user-images.githubusercontent.com/79726511/214741929-28cf1747-67cf-4bf1-934d-aa35a5a53e70.png)

Thanks

**Comment by kasperbolarsen** [comment]
If you check your crawled properties, you should see an ows_MainTypeTermTest property. This property contains the readable version of the Managed Property/Term label, and in most cases I map that property to one of the RefinableStrings as I often need that property as a filter/refiner. Try that :-)

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  thanks for the reply. i did the following:-

I define the Refinalbe as follow:-

![image](https://user-images.githubusercontent.com/79726511/214811056-53cf38ab-0db3-449d-b3bf-43e50854705c.png)


but when i am showing the refinablestring04 inside my search result, i did not get any value:-

![image](https://user-images.githubusercontent.com/79726511/214811202-8985e18d-4618-4f50-805a-f8041372d0af.png)

![image](https://user-images.githubusercontent.com/79726511/214811394-f0f66bcd-6c1c-4cbd-abeb-5603c0b23650.png)


any advise please?

not sure what is going on?
Thanks

**Reply by kasperbolarsen** [reply]
You have to wait up to 24 hours in order for that mapping to take effect 😐

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  i hope this will work after 24 hours .. let see thanks

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  till now it is not working :(

**Reply by kasperbolarsen** [reply]
Have you added Refinablestring04 to selected properties?  Step 2 would be to verify if Refinablestring04 shows up SP Query tool

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen yes it has been added since 3 days:-

![image](https://user-images.githubusercontent.com/79726511/215270518-27d11125-d086-45c5-8e58-7a6172f68140.png)

and i have also re-index the SharePoint site 3 days ago... looks there is a bug somewhere.

Second point, i have never used the SP search tool, so what i need to add inside it?

**Reply by kasperbolarsen** [reply]
Please verify using SP Query tool

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  can you advice what i need to define inside the tool? thanks

**Reply by kasperbolarsen** [reply]
Sure 🙂 Just to make sure that we know what the API is returning, I ask you to run the same query in the tool and check if Refinablestring04 is listed in the results. Perhaps https://www.techmikael.com/2014/03/debugging-managed-properties-using.html can point you in the right direction.

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen  i am not writing any query i am just displaying the RefinableString04 inside my Columns and managed properties, as shown in my original question screenshots .thanks

**Reply by kasperbolarsen** [reply]
I don't get it. You do not have any query in the Query Template field on page 1?

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen no i am just showing all documents inside the Result / source, as follow:-

![image](https://user-images.githubusercontent.com/79726511/215441224-cc8d8f47-269c-428a-a7bf-286a96f46a4c.png)

**Reply by wobba** [reply]
Your issue is the result source "Documents". It does not have `{searchTerms}`. You should use the default result source instead `LocalSharePointResults`.

**Reply by mvcsharepointdev** [reply]
@wobba @kasperbolarsen  things are not working at my end. here what i have:-

1) I have a site column with this internal name of type managed metadata:-
![image](https://user-images.githubusercontent.com/79726511/215500043-03b48297-6fc9-46fe-8aa0-df4aff340aca.png)

2) and i linked the related managed property with the RefinableString04, as follow (5 days ago):-
![image](https://user-images.githubusercontent.com/79726511/215500350-69b69a49-ac70-4515-8297-e7ce8ca6b0b7.png)


now inside my modern search web part, i did the following:-

1) settings as follow:-

![image](https://user-images.githubusercontent.com/79726511/215500557-ff0e03c1-94bf-4abb-8627-682aff81f421.png)
![image](https://user-images.githubusercontent.com/79726511/215500693-78095a81-52f3-46d4-af5a-df19512f48a6.png)

2) Define the RefinableString04 inside the Selected properties,:-

![image](https://user-images.githubusercontent.com/79726511/215500916-337e1050-a3ba-4a14-9188-47067f1adc69.png)

3) use detailed list
![image](https://user-images.githubusercontent.com/79726511/215501022-6ce180ea-c5c9-4edd-9ac7-296b6bc83966.png)

4) define this slot inside the Managed columns
![image](https://user-images.githubusercontent.com/79726511/215501217-d0591cda-6c4c-4aa2-80b0-97990fa3ef5c.png)

5) define this as a slot:-
![image](https://user-images.githubusercontent.com/79726511/215501368-f6066942-1f1e-484c-ae58-9c856427bb68.png)


now if i search for a documentt the column will be empty, as follow:-
![image](https://user-images.githubusercontent.com/79726511/215501558-5f459565-0bfb-4405-ae7a-ed84506ada25.png)


although the document has values for the managed metadata column as shown in the list view as follow:

![image](https://user-images.githubusercontent.com/79726511/215501723-c6b84b88-b259-4686-921d-01231c0bfeef.png)



any advice please?
Thanks

**Reply by mvcsharepointdev** [reply]
@wobba @kasperbolarsen can you please have a look at my latest settings .. thanks

**Comment by wobba** [comment]
You can take a look at the "localization" setting in the docs https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/sharepoint-search/

**Reply by mvcsharepointdev** [reply]
@wobba  did not get your point, how localization will fix my issue? Thanks

**Reply by wobba** [reply]
Because if the id format is correct it will load the labels. If not map the ows_colname cp to a managed property and you get the indexed display label.
